src/lib/support/BitFlags.h

```diff
@@ -87,6 +88,40 @@ class BitFlags
         return *this;
     }
 
+    /**
+     * SetField to value via a mask shifting.
+     *
+     * @param mask      Typed flag(s) to used as mask, flag(s) must be contiguous. Any flags not in @a v are unaffected.
+     * @param value     New value to apply at the field/mask location.
+     */
+    constexpr BitFlags & SetField(FlagsEnum mask, IntegerType value)
+    {
+        IntegerType bitMask = static_cast<IntegerType>(mask);
+        IntegerType shift   = GetShift(bitMask);
+
+        // Clear bits overlayed by the mask
+        mValue = static_cast<IntegerType>(mValue & ~bitMask);
+        // Set the right bits
+        mValue |= static_cast<IntegerType>(bitMask & (value << shift));
+
+        return *this;
+    }
+
+    /**
+     * GetField to value via a mask shifting.
+     *
+     * @param mask      Typed flag(s) to used as mask, flag(s) must be contiguous. Any flags not in @a v are unaffected.
+     * @returns         Value from the underlying field/mask
+     */
+    IntegerType GetField(FlagsEnum mask) const
```

Comment:
Why do we allow IntegerType to escape bit flags?

Reply:
maybe a semantic problem... a GetFieldValue.. or GetRawField. something in that spirit since we have a  SetRaw(IntegerType value)
and an IntegerType Raw()